packages/react-vapor/src/components/tab/Tab.tsx

```diff
@@ -15,50 +19,56 @@ export interface ITabOwnProps {
     url?: string;
 }
 
-export interface ITabStateProps {
-    isActive: boolean;
-}
-
-export interface ITabDispatchProps {
-    onRender: () => void;
-    onDestroy: () => void;
-    onSelect: (e: React.MouseEvent) => void;
-}
-
-export interface ITabProps extends ITabOwnProps, Partial<ITabStateProps>, Partial<ITabDispatchProps> {}
-
-export class Tab extends React.Component<ITabProps, any> {
-    render() {
-        const className = classNames('tab', {
-            enabled: !this.props.disabled,
-            disabled: this.props.disabled,
-            active: this.props.isActive,
-        });
+const enhance = connect(
+    createStructuredSelector({isActive: TabSelectors.getIsTabSelected}),
+    (dispatch: IDispatch, ownProps: ITabOwnProps) => ({
+        onRender: (): void => void dispatch(addTab(ownProps.id, ownProps.groupId)),
+        onDestroy: (): void => void dispatch(removeTab(ownProps.id, ownProps.groupId)),
+        onSelect: (e: React.MouseEvent): void => void dispatch(selectTab(ownProps.id, ownProps.groupId)),

```

Comment:
Why do we need to specify the type `void`? Will there be any type error or failing test if we remove voids?

Reply:
Since we use `typeof enhance`, every function we declare in there defines the type that will be generated for the component. So if we remove `void`, the return type will be `IReduxAction` and its not something that we want.